src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
+        if (node != null) {
+            node.value = value;
+            return;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> firstNote = table[(table.length - 1) & hash(key)];
+        Node<K,V> nextNote;
+        int hash = hash(key);
+        if (firstNote != null) {
+            if (firstNote.hash == hash
+                    && (Objects.equals(key, firstNote.key))) {
+                return firstNote;
+            }
+            if ((nextNote = firstNote.next) != null) {
+                do {
+                    if (nextNote.hash == hash
+                            && (Objects.equals(key, nextNote.key))) {
+                        return nextNote;
+                    }
+                } while ((nextNote = nextNote.next) != null);
+            }
+        }
+        return null;
+    }
+
+    private int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >>> 16);
+    }
+
+    private void resize() {
+        Node<K,V>[] oldTab = table;
+        int oldCap = oldTab.length;
+        int newCap = oldCap << 1;
+        threshold = threshold << 1;
+        @SuppressWarnings({"unchecked"})
+        Node<K,V>[] newTab = (Node<K,V>[])new Node[newCap];
+        table = newTab;
+        for (Node<K, V> bucket : oldTab) {
+            Node<K, V> node = bucket;
+            if (node != null) {
+                do {
+                    placeNode(node.key, node.value);
+                    node = node.next;
+                } while (node != null);
+            }
+        }
+    }
+
+    private Node<K,V> placeNode(K key, V value) {
+        int hash = hash(key);
+        Node<K,V>[] tab = table;
+        Node<K,V> node = tab[(tab.length - 1) & hash];
+        if (node == null) {
+            tab[(tab.length - 1) & hash] = new Node<>(hash, key, value, null);
+        } else {
+            Node<K,V> previousNode;
+            do {
+                previousNode = node;
+                if (node.hash == hash && (Objects.equals(key, node.key))) {
+                    break;
+                }
+                node = node.next;
+            } while (node != null);
+            if (node == null) {
+                previousNode.next = new Node<>(hash, key, value, null);
+            }
```

Comment:
rewrite your logic. Algorithm:
1. Find bucket by passed key.
2. Go through all nodes in this bucket:
 - if passed key and key of current node equal change value of the current node and return;
 - if the next node after the current one is null, create a new node based on the passed key and value and set it as the next element for the current one and return;